src/pymor/models/iosys.py

```diff
@@ -624,6 +625,40 @@ def to_abcde_files(self, files_basename):
         if E is not None:
             _mmwrite(Path(files_basename + '.E'), E)
 
+    def slice(self, input_indices=None, output_indices=None):
```

Comment:
It would be nice to use the special method [`__getitem__`](https://docs.python.org/3/reference/datamodel.html#object.__getitem__).

Reply:
I'd agree, in particular because it allows you to pass proper `slices` idiomatically. However, we should discuss, whether this operation is common enough and whether the user would expect `Models` to be sliceable and what the expected behavior is.
As inputs and outputs are well-defined for each `Model` (at least currently ...), we should also think about a generic implementation which does not assume knowledge of the internal structure.

Reply:
I can't comment on the commonality of this operation (I never had a use for it myself).
Regarding other `Models`, for `StationaryModels` it would make sense to slice by the number of right-hand sides and outputs. For the general, nonlinear `InstationaryModel`

$$
\begin{align*}
M(\mu) \dot{x}(t) & = f(x(t), u(t), \mu) \\
y(t) & = g(x(t), u(t), \mu)
\end{align*}
$$

the result should be an `InstationaryModel`

$$
\begin{align*}
M(\mu) \dot{x}(t) & = f(x(t), S_1 u(t), \mu) \\
y(t) & = S_2^T g(x(t), u(t), \mu)
\end{align*}
$$

where $S_1$ and $S_2$ are some columns of the identity matrix. I'm not sure how to represent $S_1$ without defining a new `Operator` class for $f$ with a restricted input parameter. At least, $S_2$ should be easy when output functionals are separated, or it can be the `ComponentProjectionOperator`.